File: src/com/merono/rlauncher/AppAdapter.java
package com.merono.rlauncher;

import android.content.Context;
import android.graphics.drawable.Drawable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.ImageView;

import java.util.List;

/** GridView adapter to show the list of all installed applications. */
public class AppAdapter extends ArrayAdapter<AppInfo> {

    private final LayoutInflater mInflater;

    public AppAdapter(Context context, List<AppInfo> apps) {
        super(context, /* Not used */ -1, apps);
        mInflater = LayoutInflater.from(context);
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        if (convertView == null) {
            convertView = mInflater.inflate(R.layout.application, parent, false);
        }

        Drawable ic = getItem(position).icon;
        ((ImageView) convertView).setImageDrawable(ic);

        return convertView;
    }
}

File: src/com/merono/rlauncher/AppAdapter.kt
package com.merono.rlauncher

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.ImageView

/** GridView adapter to show the list of all installed applications. */
class AppAdapter(context: Context, apps: List<AppInfo>) :
    ArrayAdapter<AppInfo>(context, -1, apps) {

  private val inflater = LayoutInflater.from(context)

  override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
    var view = convertView ?: inflater.inflate(R.layout.application, parent, false)

    val ic = getItem(position).icon
    (view as ImageView).setImageDrawable(ic)

    return view
  }
}
